[BUG]

### OpenIM SDK Version

3.8.2

### OpenIM Server Version

3.8.2

### SDK Platform

Web WASM SDK

### Device Model

CentOS7.6

### Operating System Version

CentOS7.6

### Bug Description and Steps to Reproduce

When build wasm sdk in 3.8.2,will report internal/interaction/ws_js.go:30:2: reading nhooyr.io/websocket/go.mod at revision v1.8.10: unknown revision v1.8.10.

### Screenshots Link

_No response_

--- a/go.mod
+++ b/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/gorilla/websocket v1.4.2
 	github.com/jinzhu/copier v0.4.0
 	github.com/pkg/errors v0.9.1
-	google.golang.org/protobuf v1.33.0
+	google.golang.org/protobuf v1.33.0 // indirect
 	gorm.io/driver/sqlite v1.5.5
 	nhooyr.io/websocket v1.8.10
 )
@@ -21,7 +21,6 @@ require (
 	github.com/patrickmn/go-cache v2.1.0+incompatible
 	golang.org/x/image v0.15.0
 	golang.org/x/sync v0.6.0
-	google.golang.org/grpc v1.62.1
 	gorm.io/gorm v1.25.10
 )
 
@@ -57,5 +56,8 @@ require (
 	golang.org/x/sys v0.18.0 // indirect
 	golang.org/x/text v0.14.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20240318140521-94a12d6c2237 // indirect
+	google.golang.org/grpc v1.62.1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace nhooyr.io/websocket => github.com/coder/websocket v1.8.10

--- a/internal/conversation_msg/message_check.go
+++ b/internal/conversation_msg/message_check.go
@@ -86,15 +86,18 @@ func (c *Conversation) checkEndBlock(ctx context.Context, conversationID string,
 	// Perform an end-of-block check if the retrieved message count is less than requested
 	if len(*list) < count {
 		if isReverse {
+			currentMaxSeq := c.maxSeqRecorder.Get(conversationID)
 			maxSeq, _, _ := c.getMaxAndMinHaveSeqList(*list)
-			log.ZDebug(ctx, "validateAndFillEndBlockContinuity", "maxSeq", maxSeq, "conversationID", conversationID)
-			if maxSeq == c.maxSeqRecorder.Get(conversationID) { // todo Replace `1` with the minimum sequence value as defined by the user or system
+			log.ZDebug(ctx, "validateAndFillEndBlockContinuity", "maxSeq", maxSeq, "conversationID", conversationID, "currentMaxSeq", currentMaxSeq)
+			// Use >= to prevent the currentMaxSeq from being updated too slowly,
+			// which could lead to misjudgments and cause repeated message fetching."
+			if maxSeq >= currentMaxSeq { // todo Replace `1` with the minimum sequence value as defined by the user or system
 				messageListCallback.IsEnd = true
 			} else {
 				lastEndSeq, _ := c.messagePullReverseEndSeqMap.Load(conversationID)
 				log.ZDebug(ctx, "validateAndFillEndBlockContinuity", "lastEndSeq", lastEndSeq, "conversationID", conversationID)
 				// If `maxSeq` is zero and `lastEndSeq` is at the maximum server sequence, this batch is fully local
-				if maxSeq == 0 && lastEndSeq == c.maxSeqRecorder.Get(conversationID) { // All messages in this batch are local messages,
+				if maxSeq == 0 && lastEndSeq >= currentMaxSeq { // All messages in this batch are local messages,
 					// and the maximum seq of the last batch of valid messages has already reached the maximum pullable seq from the server.
 					messageListCallback.IsEnd = true
 				} else {

